src/types/Todo.ts

```diff
@@ -0,0 +1,6 @@
+export interface Todo {
+  id: number;
+  userId: number;
+  title: string;
+  completed: boolean;
+}
```

Comment:
Hint: create TodoId type and use it each time, you are typing the todo's id (in your component properties or elsewherer)

```suggestion
export type TodoId = number;

export interface Todo {
  id: TodoId;
  userId: number;
  title: string;
  completed: boolean;
}
```

usage:

onTodoUpdate(id: TodoId, data: Partial<Todo>) => ...